src/main/java/jay/TaskList.java

```diff
@@ -0,0 +1,200 @@
+package jay;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import jay.command.InvalidCommandException;
+import jay.storage.DataIOException;
+import jay.storage.InvalidDataFormatException;
+import jay.storage.Storage;
+import jay.task.Task;
+
+/**
+ * Represents a list of tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+    private int taskCount;
+    private final Storage storage;
+
+    /**
+     * Constructs a TaskList object.
+     *
+     * @param fileName The name of the file to store the tasks.
+     */
+    public TaskList(String fileName) {
+        this.tasks = new ArrayList<>();
+        this.storage = new Storage("data", fileName);
+        this.taskCount = 0;
+
+        try {
+            this.loadTasks();
+        } catch (DataIOException | InvalidDataFormatException e) {
+            this.tasks.clear();
+            this.taskCount = 0;
+        }
+    }
+
+    private TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+        this.taskCount = tasks.size();
+        this.storage = new Storage("data", "wildcard");
+    }
+
+    /**
+     * Marks a Jay.task as done.
+     *
+     * @param taskNumber The index of the Jay.task need to be marked.
+     * @return The marked Jay.task.
+     * @throws InvalidCommandException If the Jay.task number is invalid.
+     * @throws DataIOException If there is an error saving the tasks.
+     */
+    public Task markAsDone(int taskNumber) throws DataIOException, InvalidCommandException {
+        this.isValidTaskNumberCheck(taskNumber);
+        this.tasks.get(taskNumber - 1).markAsDone();
+        this.saveTasks();
+        return this.tasks.get(taskNumber - 1);
+    }
+
+    /**
+     * Mark a Jay.task as not done.
+     *
+     * @param taskNumber The index of the Jay.task.
+     * @return The marked Jay.task.
+     * @throws InvalidCommandException If the Jay.task number is invalid.
+     * @throws DataIOException If there is an error saving the tasks.
+     */
+    public Task markAsNotDone(int taskNumber) throws DataIOException, InvalidCommandException {
+        this.isValidTaskNumberCheck(taskNumber);
+        this.tasks.get(taskNumber - 1).markAsNotDone();
+        this.saveTasks();
+        return this.tasks.get(taskNumber - 1);
+    }
+
+    /**
+     * Checks if the Jay.task number is valid.
+     *
+     * @param taskNumber The index of the Jay.task need to be checked.
+     * @throws InvalidCommandException If the Jay.task number is invalid.
+     */
+    private void isValidTaskNumberCheck(int taskNumber) throws InvalidCommandException {
```

Comment:
consider changing the type to boolean? Or changing the naming so that it sounds less like a boolean?